modules/tooling-support-parent/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/config/InternalDeclarationSession.java

```diff
@@ -135,6 +140,24 @@ public ValueResult getValues(ParameterizedElementDeclaration component, String p
     }
   }
 
+  @Override
+  public MetadataResult<MetadataKeysContainer> getMetadataKeys(ComponentElementDeclaration component) {
+    return artifactHelper()
+        .findComponentModel(component)
+        .map(cm -> {
+          Optional<ConfigurationInstance> configurationInstance =
+              ofNullable(component.getConfigRef()).flatMap(name -> artifactHelper().getConfigurationInstance(name));
+          MetadataKey metadataKey = buildMetadataKey(cm, component);
+          ClassLoader extensionClassLoader = getClassLoader(artifactHelper().getExtensionModel(component));
+          return withContextClassLoader(extensionClassLoader, () -> {
+            DefaultMetadataContext metadataContext = createMetadataContext(configurationInstance, extensionClassLoader);
+            return new MetadataMediator<>(cm).getMetadataKeys(metadataContext, metadataKey, new ReflectionCache());
+          });
+        })
+        .orElseGet(() -> MetadataResult.success(MetadataKeysContainerBuilder.getInstance()
```

Comment:
I would say that in this case it should be a failure as the only case for this orElse is if the componentModel is not present or found for the declaration. It would mean that they were trying to resolve keys for a component that is not defined in any of the extension models declared for the session :S

Reply:
Please review the same case for types. It should fail if component is not found